backend/users/domain/repository.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+from users.domain.converters import convert_user
+from users.domain.entities import UserID, User
+from users import models
+from django.contrib.auth.tokens import default_token_generator
+
+
+class UsersRepository:
+    def get_by_id(self, user_id: UserID) -> Optional[User]:
+        user = models.User.objects.filter(id=user_id).first()
+
+        if not user:
+            return None
+
+        return convert_user(user)
+
+    def validate_reset_password_token(self, user: User, token: str) -> bool:
+        return default_token_generator.check_token(user, token)
```

Comment:
this feels likes it doesn't really belong here

Reply:
I made the `User` entity similar enough to the django model so that `default_token_generator.check_token` can work with the entity and not the model, so I can move it outside the repository

otherwise we would need the django object to call `check_token`